ui/library/src/context/PageRenderContext.ts

```diff
@@ -0,0 +1,243 @@
+import * as React from 'react';
+import { pdfjs } from 'react-pdf';
+
+import { Nullable } from '../components/types/utils';
+import { logProviderWarning } from '../utils/provider';
+import { PageRotation } from '../utils/rotate';
+
+export type RenderState = {
+  promise: Promise<string>;
+  objectURL: Nullable<string>;
+};
+
+export type PageNumberToRenderStateMap = Map<number, RenderState>;
+
+export interface IPageRenderContext {
+  pageRenderStates: PageNumberToRenderStateMap;
+  getObjectURLForPage: (args: { pageNumber?: number; pageIndex?: number }) => Nullable<string>;
+  isBuildingObjectURLForPage: (args: { pageNumber?: number; pageIndex?: number }) => boolean;
+  buildObjectURLForPage: (args: { pageNumber?: number; pageIndex?: number }) => Promise<string>;
```

Comment:
I recall there is a type for `{ pageNumber?: number; pageIndex?: number }` in your previous PR, right?

Reply:
@yensung i will change it to that type